src/components/schedule-visualizer/tests/robot-image-icon.test.tsx

```diff
@@ -45,4 +43,31 @@ describe('RobotImageIcon', () => {
     );
     expect(root.find('circle#shadow[fill*="conflict"]')).toHaveLength(1);
   });
+
+  test('should call dispatchIconError when robotPrimaryColor throws an error', async () => {
+    jest.spyOn(colorManager, 'robotPrimaryColor').mockRejectedValue(() => 'err');
+    const mockDispatchIconError = jest.fn();
+    const root = mount(
+      <svg>
+        <RobotImageIcon
+          robot={makeRobot()}
+          footprint={1}
+          colorManager={colorManager}
+          dispatchIconError={mockDispatchIconError}
+          iconPath={iconPath}
+          fleetName={fleetName}
+        />
+      </svg>,
+    );
+    await act(
+      () =>
+        new Promise<void>((resolve) => {
+          setImmediate(() => {
+            root.update();
+            resolve();
+          });
+        }),
+    );
```

Comment:
Just in case, did you tried with the "act" inside the mount?

https://stackoverflow.com/questions/58127167/why-do-i-need-to-wrap-react-enzyme-state-changes-in-act